src/parsers/perf/perfparser.cpp

```diff
@@ -1372,6 +1379,11 @@ void PerfParser::startParseFile(const QString& path, const QString& sysroot, con
             emit callerCalleeDataAvailable(d.callerCalleeResult);
             emit eventsAvailable(d.eventResult);
             emit parsingFinished();
+
+            // abuse of function but gets the job done
+            if (d.m_numSamplesWithMoreThanOneFrame == 0) {
+                emit exportFailed(QLatin1String("Could not generate callgraph. Did you forgot --call-graph?"));
```

Comment:
Also, please make the error message something like this:

`Samples contained no call stack frames. Consider passing <code>--call-graph dwarf</code> to <code>perf record</code>.`

And I think this should just be a warning, not an error - so distinguish it accordingly when showing the message please - this may require another parameter or a second message box that's appropriately configured.

Furthermore, this should be PerfParser::tr() as we want to be able to translate this.

Reply:
that signal was already there, I only used it